src/EventStore.Projections.Core/Services/Processing/EmittedStream.cs

```diff
@@ -677,31 +682,79 @@ yield return
 					SubmitWriteEvents();
 					return;
 				}
+				
+				var report = ValidateEmittedEventInRecoveryMode(eventToWrite);
+
+				if (report is IgnoredEmittedEvent)
```

Comment:
I think it would be useful to add some verbose logging around this.

Reply:
Done